File: project_euler/Problem042/Problem042.java
package project_euler;

import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStreamReader;

public class Problem042 {
    public void solution() throws IOException {
        String fileName = "words.txt";
        int result = 0;
        try (BufferedReader handle = new BufferedReader(new InputStreamReader(new FileInputStream(fileName)))) {
            String line;
            while (null != (line = handle.readLine())) {
                int value = 0;
                for (char letter : line.toCharArray()) {
                    value += letter - 'A' + 1;
                }
                double auxiliary = Math.sqrt((value << 3) + 1);
                if (Math.abs(Math.floor(auxiliary + 0.5) - auxiliary) < 0.00000001 && 0 != ((int) auxiliary & 1)) {
                    ++result;
                }
            }
        }
        System.out.println(result);
    }
}

File: project_euler/Problem042/Problem042.js
function Problem042() {
    let fileName = 'words.txt'
    let result = 0
    const file = require('n-readlines')
    let handle = new file(fileName)
    let line
    while (line = handle.next()) {
        let value = 0
        for (let letter of line.toString()) {
            value += letter.charCodeAt(0) - 'A'.charCodeAt(0) + 1
        }
        let auxiliary = Math.sqrt((value << 3) + 1)
        if (Math.abs(Math.floor(auxiliary + 0.5) - auxiliary) < 0.00000001 && (Math.floor(auxiliary) & 1)) {
            ++result
        }
    }
    console.log(result)
}

module.exports = {
    Problem042 : Problem042
};
